controllers/automation/policyautomation_controller.go

```diff
@@ -378,10 +409,24 @@ func (r *PolicyAutomationReconciler) Reconcile(
 		} else if policyAutomation.Spec.Mode == policyv1beta1.Once {
 			log := log.WithValues("mode", string(policyv1beta1.Once))
 			targetList := common.FindNonCompliantClustersForPolicy(policy)
+
 			if len(targetList) > 0 {
+				//nolint:all
+				exist, _ := common.CheckAnsiblejobExist(policyAutomation, r.DynamicClient)
```

Comment:
You don't need to check for the existing AnsibleJob because the way the "Once" mode works, it creates an AnsibleJob and then the mode is set to disabled. If it's put back to "Once" mode, it means the user explicitly did this and wants the automation to be rerun on the next triggering event.

Reply:
I think suggested this, because there had been an issue in the tests where the reconcile was triggered twice, and the second time wasn't "seeing" that the mode had been set to disabled. It would also be possible in general for the AnsibleJob to be created, but the update to the automation (setting it to disabled) might not succeed for some other reason, meaning two AnsibleJobs would be created for one `Once` event.

Reply:
This is the part that Justin help me with. I guess updating mode status takes a while. Therefore, when it falls to second event it is still mode=once so it creates second ansiblejob

Reply:
In that case, I'd prefer to "patch" instead of "update" the PolicyAutomation object when setting this to disabled to significantly reduce the change of an update failing. Otherwise, you'd need to be able to distinguish between a duplicate AnsibleJob trigger versus a user resetting the mode back to "Once".

Reply:
I tried patch same result

Reply:
@yiraeChristineKim what was the result you saw? The reconcile ran again?

Reply:
so update disable doesn't work for the second event. the second event is supposed not to fall in once. In the second event update failed(error message changed something..).  As a result, randomly it creates 1 ansiblejob or 2 ansiblejobs

Reply:
I added sleep for 100 milliseconds.. this works well

Reply:
@yiraeChristineKim what if you updated `policyAuomtationPredicateFuncs` so that when the mode is disabled, it doesn't trigger a reconcile?

Reply:
same result, first try e2e test, it creates one ansiblejob but second try, it creates 2 ansiblejobs

Reply:
After thinking about this, it's not the PolicyAutomation causing the reconcile in the case you're fixing but an additional replicated policy causing the reconcile.

Reply:
oh wait what I read it  **This is the equivalent of calling Watches** https://github.com/kubernetes-sigs/controller-runtime/blob/v0.14.6/pkg/builder/controller.go#L83 i thought watch for same